src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/BlobDispatcherTask.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.blobrouter.tasks;
+
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import org.slf4j.Logger;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.tasks.processors.ContainerProcessor;
+import uk.gov.hmcts.reform.blobrouter.util.StorageClientManager;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class BlobDispatcherTask {
+
+    static final String TASK_NAME = "blob-dispatcher";
+
+    private static final Logger LOGGER = getLogger(BlobDispatcherTask.class);
+
+    private final BlobServiceAsyncClient storageClient;
+    private final ContainerProcessor containerProcessor;
+    private final ServiceConfiguration serviceConfiguration;
+
+    public BlobDispatcherTask(
+        BlobServiceAsyncClient storageClient,
+        ContainerProcessor containerProcessor,
+        ServiceConfiguration serviceConfiguration
+    ) {
+        this.containerProcessor = containerProcessor;
+        this.serviceConfiguration = serviceConfiguration;
+        this.storageClient = storageClient;
+    }
+
+    public void run() {
+        LOGGER.info("Started {} job", TASK_NAME);
+
+        StorageClientManager
+            .getAvailableContainers(storageClient, serviceConfiguration)
+            .toStream()
+            .map(BlobContainerItem::getName)
+            .map(storageClient::getBlobContainerAsyncClient)
+            .forEach(containerProcessor::process);
```

Comment:
What we're doing here is:
- get existing containers, but only those that are enabled in configuration
- for each of those containers get a client
- using the appropriate client, process each container

While technically it does the job, I'd recommend an alternative approach:
- get the names of all enabled containers in configuration
- for each of those names, process the container of a given name

There are two benefits for the second approach:
- When the container from the configuration doesn't exist, we'll get an error. In current approach, we don't know about the problem.
- This task can just use a `containerProcessor` and not `storageClient`. `containerProcessor` could use `storageClient` directly, and these lower-level details could be hidden from the task. That would make it easier to test.

Reply:
Updated the pr as suggested